### PR TITLE
Exercises Page Complete

### PR DESCRIPTION
**Pull request by Eddddiiiieeee**

### CODE REVIEW

**Review by EmreCem — APPROVED** [review]
- Page and tests are fine

- Thanks for applying changes to the coupons page as well

- I am approving the merge but you could maybe add more comments to the cypress test files. Much less commenting in there compared to your actual code (maybe try to fit the 15% requirement Theo was talking about)

All in all, I really like your coding style :)

File: src/Components/GridComponents.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  )
+}
+
+/**
+ * The properties used by the CustomToolbar component.
+ * Boolean value properties that, when true, enables the
+ * rendering of the corresponding buttons in the DataGrid.
+ */
+interface ToolbarProps {
+  density?: true | undefined
```
Comment (nstylo) [inline src/Components/GridComponents.tsx:37]: undefined is implied through '?' I believe?

File: src/Components/MissedIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
+import React from 'react'
```
Comment (nstylo) [inline src/Components/MissedIcon.tsx:1]: why not use an svg directly?

Comment (Eddddiiiieeee) [reply src/Components/MissedIcon.tsx:1]: I tried doing it directly, but somehow it didn't work. so I had to convert to JSX and used that as a workaround.
However, I just tried it out again and now it works. This is now a direct svg in the latest commit.

**Pull request merged**